File: cpp/hamming/hamming.cpp
#include "hamming.h"

namespace hamming {
  int compute(std::string a, std::string b) {
    int count = 0;
    if (a.size() != b.size()) {
      throw std::domain_error("unequal length");
    }
    for(std::string::size_type i = 0; i < a.size(); i++) {
      if (a[i] != b[i]) {
        count++ ;
      }
    }
    return count;
  }
}

File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand_a, strand_b)
    raise ArgumentError if strand_a.chars.length != strand_b.chars.length
    strand_a.chars.zip(strand_b.chars).select{|a,b| a!=b}.count
  end
end

module BookKeeping
  VERSION = 3
end
